lib/vtls/sectransp.c

```diff
@@ -138,6 +139,474 @@ struct ssl_backend_data {
+ We make internal data consistent and based on TLS names, i.e. all st_cipher
+ item names start with "TLS_" prefix.
+ Summarizing all above, those 28 first ciphers are presented in our table with
+ both TLS and SSL names. Their cipher numbers are assigned based on SDK enum
+ value for SSL cipher, which matches to IANA TLS number.
+ */
+#define CIPHER_DEF_SSLTLS(num_wo_prefix, weak) \
+  { "TLS_" #num_wo_prefix, "SSL_" #num_wo_prefix, SSL_##num_wo_prefix, weak }
+
+/*
+ Cipher suites were marked as weak based on the following:
+ RC4 encryption - rfc7465, the document contains list of deprecated ciphers.
+     Marked in the code below as weak.
+ RC2 encryption - many mentions, was found vulnerable to a relatively easy
+     attack https://link.springer.com/chapter/10.1007%2F3-540-69710-1_14
+     Marked in the code below as weak.
+ DES and IDEA encryption - rfc5469, has list of deprecated ciphers.
+     Marked in the code below as weak.
+ Anonymous Deffie-Hellman authentication and anonymous elliptic curve
+     Deffie-Hellman - vulnerable to man-in-the-middle attack. Deprecated by
+     RFC 4346 aka TLS 1.1 (section A.5, page 60)
+ Null bulk encryption suites - not encrypted communication
+ Export ciphers, i.e. ciphers with restrictrions to be used outside US for
+     software exported to some countries, they were excluded from TLS 1.1
+     version. More precisely, they were noted as ciphers which MUST NOT be
+     negotiated in RFC 4346 aka TLS 1.1 (section A.5, pages 60 and 61).
+     All of of those filters were considered weak because they contain weak
+     algorithm like DES, RC2 or RC4 and already considered weak by another
+     criterion.
+
+ 3DES - NIST deprecated it and going to retire it by 2023
+ https://csrc.nist.gov/News/2017/Update-to-Current-Use-and-Deprecation-of-TDEA
+     OpenSSL https://www.openssl.org/blog/blog/2016/08/24/sweet32/ also
+     deprecated those ciphers. Some other libraries also consider it
+     vulnerable or at least not strong enough.
+     We DID NOT mark them in the code below as weak because it is a
+     recommendation only, and due to backward compatibility with previous
+     implementation. We highlighted them in code and made ready for easy flip
+     to weak.
+
+ CBC ciphers are vulnerable with SSL3.0 and TLS1.0:
+ https://www.cisco.com/c/en/us/support/docs/security/email-security-appliance
+ /118518-technote-esa-00.html
+     We don't take care about this issue because it is resolved by later
+     TLS versions and for us it requires more complicated checks, we need
+     to check a protocol version also. Vulnerability doesn't look very
+     critical and we do not filter out those suites.
+ */
+
+#define CIPHER_WEAK_NOT_ENCRIPTED   TRUE
+#define CIPHER_WEAK_RC_ENCRIPTION   TRUE
+#define CIPHER_WEAK_DES_ENCRIPTION  TRUE
+#define CIPHER_WEAK_IDEA_ENCRIPTION TRUE
+#define CIPHER_WEAK_ANON_AUTH       TRUE
+#define CIPHER_STRONG_ENOUGH        FALSE
+#define CIPHER_STRONG_ARGUABLY_3DES FALSE
```

Comment:
I'd be in favor of disabling 3DES cipher-suites by default. 3DES is vulnerable to the Sweet32 attack.